spec/lib/html_processors/amp_iframe_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'html_processor'
+require 'html_processor/amp_iframe'
+
+RSpec.describe HTMLProcessor::AmpIframe do
+  let(:processor) { described_class.new(html) }
+
+  describe '.process' do
+    subject { Nokogiri::XML(processed_html.strip).children.first }
+    let(:attributes) { subject.attributes.transform_values!(&:value) }
+
+    let(:processed_html) { processor.process(HTMLProcessor::NON_VIDEO_IFRAME) }
+
+    let(:xpath) do
+      '//iframe[not(starts-with(@src, "https://www.youtube.com/embed"))]'
+    end
+
+    context 'when attributes are present' do
+      let(:html) do
+        <<-EOHTML
+          <iframe
+            src="https://google.com"
+            alt="test_alt"
+            height="test_height"
+            width="test_width"
+            srcdoc="test_srcdoc"
+            frameborder="test_frameborder"
+            allowfullscreen="test_allowfullscreen"
+            allowpaymentrequest="test_allowpaymentrequest"
+            allowtransparency="test_allowtransparency"
+            referrerpolicy="test_referrerpolicy"
+            sandbox="test_sandbox" />
+        EOHTML
+      end
+
+      it 'copies the attributes across correctly from the original iframe' do
```

Comment:
Can you rename to something like: "copies original iframe attributes to ..." and then insert the destination, please?